apps/backend/src/groups/groups.service.ts

```diff
@@ -93,4 +112,38 @@ export class GroupsService {
 
     return groupToDelete;
   }
+
+  syncUserGroups(user: User, groups: Array<string>) {
+    // Check if user is any existing groups that they should not be
+    user.$get('groups', {include: [User]}).then((currentGroups) => {
+      currentGroups.filter(group => !groups.includes(group.name)).forEach((groupToLeave) => {
```

Comment:
https://2ality.com/2021/01/looping-over-arrays.html

prefer `for of` over `foreach`